In [1]:
import pandas as pd
import numpy as np
from sklearn.svm import SVC
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import RobustScaler
from sklearn.metrics import accuracy_score

In [2]:
df = pd.read_csv('diabetes.csv')

In [3]:
df

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50        1  
1                       0.351   31        0  
2                       0.672   32        1  
3                       0.167   21        0  
4                       2.288   33        1  
..                        ...  ...      ...  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0  

[768 rows x 9 columns]

In [4]:
X = df.iloc[:,:-1].values
y = df.iloc[:,-1].values

In [5]:
rs = RobustScaler()
X = rs.fit_transform(X)

In [6]:
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.2,random_state=0)

In [7]:
print(X_train.shape)
print(X_test.shape)
print(y_train.shape)
print(y_test.shape)

(614, 8)
(154, 8)
(614,)
(154,)


In [8]:
lst = []
for i in range(1,50):
    svm = SVC(kernel='linear',C=i)
    svm.fit(X_train,y_train)
    a = accuracy_score(y_true=y_test,y_pred=svm.predict(X_test))
    lst.append(round(a*100,3))

In [9]:
for i in range(len(lst)):
    if lst[i]==max(lst):
        print(f'for c = {i+1} and accuracy = {lst[i]} %')

for c = 1 and accuracy = 82.468 %
for c = 2 and accuracy = 82.468 %
for c = 3 and accuracy = 82.468 %
for c = 4 and accuracy = 82.468 %
for c = 5 and accuracy = 82.468 %
for c = 6 and accuracy = 82.468 %
for c = 7 and accuracy = 82.468 %
for c = 8 and accuracy = 82.468 %
for c = 9 and accuracy = 82.468 %
for c = 10 and accuracy = 82.468 %
for c = 11 and accuracy = 82.468 %


## For all above values of C, maximum accuracy is achived which is 82.468%